### PR TITLE
Rating

### PR DESCRIPTION
**Pull request by lwmcdona**
Added functionality for rating. Has yet to be full tested.

### CODE REVIEW

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 1]
```diff
@@ -80,6 +84,11 @@
     private MaterialButton addNewBookImageButton;
     private String isbn_code;
     private ProgressDialog progressDialog;
+    private Dialog ratingDialog;
+
+    //TODO: DELETE
+    private Button tmpRating;
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]: Should this be deleted?

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 2]
```diff
@@ -110,6 +119,20 @@ protected void onCreate(Bundle savedInstanceState) {
         setBookDescriptionFieldsEditable(false);
 
         getIncomingIntents();
+
+        //TODO:DELETE
+        tmpRating = findViewById(R.id.tmp_rate);
+        tmpRating.setOnClickListener(new View.OnClickListener() {
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]: Should this be deleted as well?
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java (hunk 1 above)
Comment (tymoorej) [reply app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 3]
```diff
@@ -80,6 +84,11 @@
     private MaterialButton addNewBookImageButton;
     private String isbn_code;
     private ProgressDialog progressDialog;
+    private Dialog ratingDialog;
+
+    //TODO: DELETE
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 4]
```diff
@@ -110,6 +119,20 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(ViewBookActivity.this, RateUserActivity.class);
+                intent.putExtra(UNAME,book.getOwnerUserName());
+//                intent.putExtra(UBO,UserIdentity.OWNER);
+                intent.putExtra(UBO,UserIdentity.BORROWER);
+//                startActivityForResult(intent,RateUserActivity.RATEOWNER);
+                startActivityForResult(intent,RateUserActivity.RATEBORROWER);
+            }
+        });
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 5]
```diff
@@ -110,6 +119,20 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        tmpRating.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(ViewBookActivity.this, RateUserActivity.class);
+                intent.putExtra(UNAME,book.getOwnerUserName());
+//                intent.putExtra(UBO,UserIdentity.OWNER);
+                intent.putExtra(UBO,UserIdentity.BORROWER);
+//                startActivityForResult(intent,RateUserActivity.RATEOWNER);
+                startActivityForResult(intent,RateUserActivity.RATEBORROWER);
+            }
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 6]
```diff
@@ -110,6 +119,20 @@ protected void onCreate(Bundle savedInstanceState) {
         setBookDescriptionFieldsEditable(false);
 
         getIncomingIntents();
+
+        //TODO:DELETE
+        tmpRating = findViewById(R.id.tmp_rate);
+        tmpRating.setOnClickListener(new View.OnClickListener() {
+            @Override
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 7]
```diff
@@ -110,6 +119,20 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            public void onClick(View v) {
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 8]
```diff
@@ -110,6 +119,20 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                Intent intent = new Intent(ViewBookActivity.this, RateUserActivity.class);
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 9]
```diff
@@ -110,6 +119,20 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                intent.putExtra(UNAME,book.getOwnerUserName());
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 10]
```diff
@@ -110,6 +119,20 @@ protected void onCreate(Bundle savedInstanceState) {
...
+//                intent.putExtra(UBO,UserIdentity.OWNER);
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 11]
```diff
@@ -110,6 +119,20 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                intent.putExtra(UBO,UserIdentity.BORROWER);
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 12]
```diff
@@ -110,6 +119,20 @@ protected void onCreate(Bundle savedInstanceState) {
...
+//                startActivityForResult(intent,RateUserActivity.RATEOWNER);
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 13]
```diff
@@ -110,6 +119,20 @@ protected void onCreate(Bundle savedInstanceState) {
         setBookDescriptionFieldsEditable(false);
 
         getIncomingIntents();
+
+        //TODO:DELETE
+        tmpRating = findViewById(R.id.tmp_rate);
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 14]
```diff
@@ -110,6 +119,20 @@ protected void onCreate(Bundle savedInstanceState) {
         setBookDescriptionFieldsEditable(false);
 
         getIncomingIntents();
+
+        //TODO:DELETE
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

File: app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java [hunk 15]
```diff
@@ -110,6 +119,20 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        tmpRating = findViewById(R.id.tmp_rate);
+        tmpRating.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(ViewBookActivity.this, RateUserActivity.class);
+                intent.putExtra(UNAME,book.getOwnerUserName());
+//                intent.putExtra(UBO,UserIdentity.OWNER);
+                intent.putExtra(UBO,UserIdentity.BORROWER);
+//                startActivityForResult(intent,RateUserActivity.RATEOWNER);
+                startActivityForResult(intent,RateUserActivity.RATEBORROWER);
```
Comment (tymoorej) [inline app/src/main/java/com/stl/skipthelibrary/Activities/ViewBookActivity.java]:
```suggestion
```

**Comment by QFD-Felix** [comment]
bug has been fixed, however, the functionality couldn't be tested completely, since couldn't reach the final handoff state in the emulator

**Pull request merged**